src/gc/pooltable.d

```diff
@@ -0,0 +1,287 @@
+/**
+ * A sorted array to quickly lookup pools.
+ *
+ * Copyright: Copyright Digital Mars 2001 -.
+ * License:   $(WEB www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+ * Authors:   Walter Bright, David Friedman, Sean Kelly, Martin Nowak
+ */
+module gc.pooltable;
+
+static import cstdlib=core.stdc.stdlib;
+
+struct PoolTable(Pool)
+{
+    import core.stdc.string : memmove;
+
+nothrow:
+    void reset()
+    {
+        cstdlib.free(pools);
+        pools = null;
+        npools = 0;
+    }
+
+    bool insert(Pool* pool)
+    {
+        auto newpools = cast(Pool **)cstdlib.realloc(pools, (npools + 1) * pools[0].sizeof);
```

Comment:
>But then we couldn't explicitly destroy Pools after a collection.

Have you ever seen this happen in minimize()? It seems very unlikely to me.

> I'd generally avoid using GC memory for the metadata as it creates complicated corner cases.

Yeah, I didn't mean to make immediate changes, but just pondering possibilities ;-)